modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -1989,6 +2007,19 @@ else if (active) {
 		return objectDefinition;
 	}
 
+	private void _updateObjectDefinitionPortlet(
+			ObjectDefinition objectDefinition, long rootObjectDefinitionId)
+		throws PortalException {
+
+		if ((objectDefinition.getObjectDefinitionId() !=
+				rootObjectDefinitionId) &&
```

Comment:
Hi @larissacribeiro use `objectDefinition.isRootDescedant()` instead of `objectDefinition.getObjectDefinitionId() != rootObjectDefinitionId` and sort, isPortlet should comes before, for example:

```java
if (objectDefinition.isPortlet() && objectDefinition.isRootDescedant()) {
```

Thank you!

Reply:
Just updated it, thanks! I've had to set the values of the objects definitions again, after changing the  rootObjectDefinitionId, or else the validation isRootDescedant() would return false, due to the id not being updated yet at the moment it was checked.